x-pack/plugins/ingest_manager/public/plugin.ts

```diff
@@ -78,18 +75,19 @@ export class IngestManagerPlugin
   }
 
   public async start(core: CoreStart): Promise<IngestManagerStart> {
-    try {
-      const permissionsResponse = await core.http.get(appRoutesService.getCheckPermissionsPath());
-      if (permissionsResponse.success) {
-        const { isInitialized: success } = await core.http.post(setupRouteService.getSetupPath());
-        return { success, registerDatasource };
-      } else {
-        throw new Error(permissionsResponse.error);
-      }
-    } catch (error) {
+    const permissionsResponse = await core.http.get<CheckPermissionsResponse>(
+      appRoutesService.getCheckPermissionsPath()
+    );
+    if (permissionsResponse.success) {
+      const successPromise = core.http
+        .post<PostIngestSetupResponse>(setupRouteService.getSetupPath())
+        .then(({ isInitialized }: { isInitialized: boolean }) => Promise.resolve(isInitialized))
+        .catch(Promise.reject);
```

Comment:
is this `.catch()` needed? Does not seem to really do anything other than to re-rewrap the error in a reject